frontend/src/tests/HealthCheck.test.tsx

```diff
@@ -0,0 +1,17 @@
+import { expect, test } from 'vitest';
+import { render, screen, waitFor } from '@testing-library/react';
+import HealthCheck from '../components/HealthCheck';
+
+test('HealthCheck heading', () => {
+  render(<HealthCheck />);
+  expect(screen.getByRole('heading', { level: 1, name: 'Health Check' })).toBeDefined();
+});
+
+test('HealthCheck paragraph stats', async () => {
+  render(<HealthCheck />);
+  await waitFor(() => {
+    expect(
+      screen.getByRole('heading', { level: 2, name: 'NMP API is healthy and ready!' }),
+    ).toBeDefined();
+  });
+});
```

Comment:
Needs more investigation to get working 100%.

The value in testing is that I expect this test to **always** pass. If I updated something in x,y,z component and this test randomly started failing, I would be alerted that my work had some unintended side effects else where in the app. many pipelines will run test as a pre-merge check as well, so bad code gets caught before being in a production environment.

Reply:
If you could test it out again, that would be awesome! I've fixed what I had to make it pass everytime and wouldn't need to do extra work like running the container to make it pass :)